frontend/apps/web/components/site-header/Upgrade.tsx

```diff
@@ -36,47 +51,69 @@ export default function Upgrade(props: UpgradeProps): ReactElement | null {
   if (!isNeosyncCloud) {
     return <UpgradeButton href={calendlyLink} target="_blank" />;
   }
+
   if (isLoading || isAccountStatusValidResp?.isValid) {
     return null;
   }
+
   const billingHref = `/${account?.name}/settings/billing`;
+
   return (
     <div className="flex flex-row gap-1 items-center">
-      <UpgradeInfoDialog
-        upgradeHref={billingHref}
-        reason={isAccountStatusValidResp?.reason}
-      />
-      <UpgradeButton href={billingHref} />
+      {!isAccountStatusValidResp?.isValid ? (
+        <UpgradeInfoDialog
+          upgradeHref={billingHref}
+          reason={isAccountStatusValidResp?.accountStatus}
+          description={isAccountStatusValidResp?.reason}
+          count={count}
+          allowedRecords={allowedRecords}
```

Comment:
A request to the `isAccountStatusValid` endpoint is being made in this component, so the values can be pulled from here.

OTOH, you could just pass the result down from the parent component since it seems it is being called up there and passing down some of its data to this component.

If you did that, this component would become less smart, which is nice :)

Reply:
woops totally missed that - yes definitely pass from parent!